hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/RangeKeysGenerator.java

```diff
@@ -0,0 +1,176 @@
+  private int range;
+
+  @CommandLine.Option(names = {"-s", "--key-start-index"},
+          description = "Start index of key.",
+          defaultValue = "0")
+  private int startIndex;
+
+
+  @CommandLine.Option(names = {"-k", "--key-encode"},
+          description = "The algorithm to generate key names. " +
+                  "Options are pureIndex, md5",
+          defaultValue = "md5")
+  private String encodeFormat;
+
+  @CommandLine.Option(names = {"-g", "--size"},
+          description = "Generated object size (in bytes) " +
+                  "to be written.",
+          defaultValue = "1")
+  private int writeSizeInBytes;
+
+  @CommandLine.Option(names = {"--clients"},
+          description =
+                  "Number of clients, defaults 1.",
+          defaultValue = "1")
+  private int clientsCount = 1;
+
+  @CommandLine.Option(
+          names = "--debug",
+          description = "Enable debugging message.",
+          defaultValue = "false"
+  )
+  private boolean debug;
+
+
+  @CommandLine.Option(
+          names = "--om-service-id",
+          description = "OM Service ID"
+  )
+  private String omServiceID = null;
+  private KeyGeneratorUtil kg;
+  private OzoneClient[] rpcClients;
+  private byte[] keyContent;
+  private Timer timer;
+
+
+  @Override
+  public Void call() throws Exception {
+    init();
+    OzoneConfiguration ozoneConfiguration = createOzoneConfiguration();
+    rpcClients = new OzoneClient[clientsCount];
+    for (int i = 0; i < clientsCount; i++) {
+      rpcClients[i] = createOzoneClient(omServiceID, ozoneConfiguration);
+    }
+
+    ensureVolumeAndBucketExist(rpcClients[0], volumeName, bucketName);
+    if (writeSizeInBytes >= 0) {
+      keyContent = RandomUtils.nextBytes(writeSizeInBytes);
+    }
+    timer = getMetrics().timer("key-read-write");
+
+    kg = new KeyGeneratorUtil();
+    runTests(this::generateRangeKeys);
+    for (int i = 0; i < clientsCount; i++) {
+      if (rpcClients[i] != null) {
+        rpcClients[i].close();
+      }
+    }
+
+    return null;
+  }
+
+  public void generateRangeKeys(long count) throws Exception {
+    int clientIndex = (int)(count % clientsCount);
+    OzoneClient client = rpcClients[clientIndex];
+    int start = startIndex + (int)count * range;
+    int end = start + range;
+
+    timer.time(() -> {
+      switch (encodeFormat) {
+      case PURE_INDEX:
+        loopRunner(kg.pureIndexKeyNameFunc(), client, start, end);
+        break;
+      case MD5:
+        loopRunner(kg.md5KeyNameFunc(), client, start, end);
+        break;
+      default:
+        loopRunner(kg.md5KeyNameFunc(), client, start, end);
+        break;
+      }
+      return null;
+    });
+  }
+
+
+  public void loopRunner(Function<Integer, String> f, OzoneClient client,
+                         int start, int end) throws Exception {
+    OzoneBucket ozbk = client.getObjectStore().getVolume(volumeName)
+            .getBucket(bucketName);
+
+    String keyName;
```

Comment:
nit: `keyName` can be declared inside the loop.

Reply:
ha! you're absolutely right! (I originally wanted to reduce the amount of new String created, but java String is immutable! -_-